scmail.py

```diff
@@ -134,42 +143,61 @@ def register(ctx, fingerprint):
 
 
 @click.command()
-@click.option('--fingerprint', '-f', prompt='Enter fingerprint of mailbox', required=True, help='The fingerprint of the yourself mailbox.')
-@click.option('--sender-fingerprint', '-s', prompt='Enter the fingerprint of sender', default='', help='The senders fingerprint.')
-@click.password_option('--password', '-p', prompt='Enter password of private key', help='The passphrase of private key')
+@click.option('--fingerprint', '-f', prompt='Enter fingerprint of mailbox',
+              required=True, type=str, help='The fingerprint of the yourself mailbox.')
+@click.option('--sender-fingerprint', '-s', prompt='Enter the fingerprint of sender',
+              type=str, help='The senders fingerprint.')
+@click.password_option('--password', '-p', prompt='Enter password of private key',
+                       help='The passphrase of private key')
 @click.pass_context
 def retrieve(ctx, fingerprint, sender_fingerprint, password):
     """Retrieve and Post messages from API"""
+    # Prepare send data.
     logging.debug(fingerprint, sender_fingerprint)
     payload = {'fingerprint': fingerprint}
     if sender_fingerprint:
         payload.update({'sender_fingerprint': sender_fingerprint})
-    # Retrieve
-    r = requests.post(SECUREMAILBOX_URL + '/retrieve/', json=payload)
+
+    # Retrieve from api.
+    r = requests.post(SECUREMAILBOX_URL + '/retrieve/', json=data)
```

Comment:
I think this may be a merge conflict that wasn't resolved correctly.

The following line should now be:
```suggestion
    r = requests.post(SECUREMAILBOX_URL + '/retrieve/', json=payload)
```

Reply:
Thanks, now it is correct.